server/lib/security/expense.ts

```diff
@@ -228,113 +228,123 @@ const getGroupedExpensesStats = (expenses: Array<Expense>) => {
 /**
  * Get some stats about the collectives of the expenses. Returns an object with the expense id as key and the stats as value.
  */
-// const getExpensesAmountsStats = async (
-//   expenses: Array<Expense>,
-//   displayCurrency,
-// ): Promise<Record<string, ExpenseAmountStats>> => {
-//   const collectiveIds = uniq(expenses.map(e => e.CollectiveId));
-//   const expenseIds = uniq(expenses.map(e => e.id));
-//   const result: Array<ExpenseAmountStats> = await sequelize.query(
-//     `
-//       WITH all_expenses AS (
-//         SELECT
-//           e.*,
-//           e.currency != :displayCurrency AS "isConvertedCurrency",
-//           CASE
-//             -- Simple case: expense is already in display currency
-//             WHEN e.currency = :displayCurrency THEN e.amount
-//             -- Convert expense to display currency
-//             ELSE e.amount * COALESCE((
-//                 SELECT rate
-//                 FROM "CurrencyExchangeRates" r
-//                 WHERE r."from" = e.currency
-//                 AND r."to" = :displayCurrency
-//                 AND r."createdAt" <= e."createdAt"
-//                 ORDER BY e."createdAt" DESC -- Most recent rate that is older than the expense
-//                 LIMIT 1
-//               ), (
-//                 -- Fix for old expenses where we didn't have an exchange rate stored yet: just use the oldest rate available
-//                 SELECT rate
-//                 FROM "CurrencyExchangeRates" r
-//                 WHERE r."from" = e.currency
-//                 AND r."to" = :displayCurrency
-//                 ORDER BY e."createdAt" ASC -- Oldest rate
-//                 LIMIT 1
-//             ))
-//             END AS "amountInHostCurrency"
-//         FROM "Expenses" e
-//         WHERE e."deletedAt" IS NULL
-//         AND (e.id IN (:expenseIds) OR e."CollectiveId" IN (:collectiveIds))
```

Comment:
1st change here: no need to filter with `OR`; filtering on collective ID already covers everything. Saves ~2000 complexity.